File: Archiver/CS/clsARCHIVEHIST.cs
// VBConversions Note: VB project level imports
using System.Collections.Generic;
using System;
using System.Drawing;
using System.Linq;
using System.Diagnostics;
using System.Data;
using Microsoft.VisualBasic;
using MODI;
using System.Xml.Linq;
using System.Collections;
using System.Windows.Forms;
// End of VB project level imports

using System.Configuration;
using System.Data.SqlClient;
using System.IO;
using System.Data.Sql;


namespace EcmArchiveClcSetup
{
	public class clsARCHIVEHIST
	{
		
		//** DIM the selected table columns
		clsDatabase DB = new clsDatabase();
		clsDma DMA = new clsDma();
		clsLogging LOG = new clsLogging();
		clsUtility UTIL = new clsUtility();
		
		string ArchiveID = "";
		string ArchiveDate = "";
		string NbrFilesArchived = "";
		string UserGuid = "";
		
		
		//** Generate the SET methods
		public void setArchiveid(ref string val)
		{
			if (val.Length == 0)
			{
				val = Guid.NewGuid().ToString();
			}
			if (val.Length == 0)
			{
				MessageBox.Show("SET: Field \'Archiveid\' cannot be NULL.: 2300");
				return;
			}
			val = UTIL.RemoveSingleQuotes(val);
			ArchiveID = val;
		}
		
		public void setArchivedate(ref string val)
		{
			val = UTIL.RemoveSingleQuotes(val);
			ArchiveDate = val;
		}
		
		public void setNbrfilesarchived(ref string val)
		{
			if (val.Length == 0)
			{
				val = "null";
			}
			val = UTIL.RemoveSingleQuotes(val);
			NbrFilesArchived = val;
		}
		
		public void setUserguid(ref string val)
		{
			val = UTIL.RemoveSingleQuotes(val);
			UserGuid = val;
		}
		
		
		
		//** Generate the GET methods
		public string getArchiveid()
		{
			if (ArchiveID.Length == 0)
			{
				MessageBox.Show("GET: Field \'Archiveid\' cannot be NULL: 2400.");
				return "";
			}
			return UTIL.RemoveSingleQuotes(ArchiveID);
		}
		
		public string getArchivedate()
		{
			return UTIL.RemoveSingleQuotes(ArchiveDate);
		}
		
		public string getNbrfilesarchived()
		{
			if (NbrFilesArchived.Length == 0)
			{
				NbrFilesArchived = "null";
			}
			return NbrFilesArchived;
		}
		
		public string getUserguid()
		{
			return UTIL.RemoveSingleQuotes(UserGuid);
		}
		
		
		
		//** Generate the Required Fields Validation method
		public bool ValidateReqData()
		{
			if (ArchiveID.Length == 0)
			{
				return false;
			}
			return true;
		}
		
		
		//** Generate the Validation method
		public bool ValidateData()
		{
			if (ArchiveID.Length == 0)
			{
				return false;
			}
			return true;
		}
		
		
		//** Generate the INSERT method
		public bool Insert()
		{
			bool b = false;
			string s = "";
			s = s + " INSERT INTO ArchiveHist(";
			s = s + "ArchiveID,";
			s = s + "ArchiveDate,";
			s = s + "NbrFilesArchived,";
			s = s + "UserGuid) values (";
			s = s + "\'" + ArchiveID + "\'" + ",";
			s = s + "\'" + ArchiveDate + "\'" + ",";
			s = s + NbrFilesArchived + ",";
			s = s + "\'" + UserGuid + "\'" + ")";
			return DB.ExecuteSqlNewConn(s, false);
			
		}
		
		
		//** Generate the UPDATE method
		public bool Update(string WhereClause)
		{
			bool b = false;
			string s = "";
			
			if (WhereClause.Length == 0)
			{
				return false;
			}
			
			s = s + " update ArchiveHist set ";
			s = s + "ArchiveID = \'" + getArchiveid() + "\'" + ", ";
			s = s + "ArchiveDate = \'" + getArchivedate() + "\'" + ", ";
			s = s + "NbrFilesArchived = " + getNbrfilesarchived() + ", ";
			s = s + "UserGuid = \'" + getUserguid() + "\'";
			WhereClause = (string) (" " + WhereClause);
			s = s + WhereClause;
			return DB.ExecuteSqlNewConn(s, false);
		}
		
		
		//** Generate the SELECT method
		public SqlDataReader SelectRecs()
		{
			bool b = false;
			string s = "";
			SqlDataReader rsData;
			s = s + " SELECT ";
			s = s + "ArchiveID,";
			s = s + "ArchiveDate,";
			s = s + "NbrFilesArchived,";
			s = s + "UserGuid ";
			s = s + " FROM ArchiveHist";
			
			string CS = DB.getGateWayConnStr(modGlobals.gGateWayID);
			SqlConnection CONN = new SqlConnection(CS);
			CONN.Open();
			SqlCommand command = new SqlCommand(s, CONN);
			rsData = command.ExecuteReader();
			return rsData;
		}
		
		
		//** Generate the Select One Row method
		public SqlDataReader SelectOne(string WhereClause)
		{
			bool b = false;
			string s = "";
			SqlDataReader rsData;
			s = s + " SELECT ";
			s = s + "ArchiveID,";
			s = s + "ArchiveDate,";
			s = s + "NbrFilesArchived,";
			s = s + "UserGuid ";
			s = s + " FROM ArchiveHist";
			s = s + WhereClause;
			
			string CS = DB.getGateWayConnStr(modGlobals.gGateWayID);
			SqlConnection CONN = new SqlConnection(CS);
			CONN.Open();
			SqlCommand command = new SqlCommand(s, CONN);
			rsData = command.ExecuteReader();
			return rsData;
		}
		
		
		//** Generate the DELETE method
		public bool Delete(string WhereClause)
		{
			bool b = false;
			string s = "";
			
			if (WhereClause.Length == 0)
			{
				return false;
			}
			
			WhereClause = (string) (" " + WhereClause);
			
			s = " Delete from ArchiveHist";
			s = s + WhereClause;
			
			b = DB.ExecuteSqlNewConn(s, false);
			return b;
			
		}
		
		
		//** Generate the Zeroize Table method
		public bool Zeroize()
		{
			bool b = false;
			string s = "";
			
			s = s + " Delete from ArchiveHist";
			
			b = DB.ExecuteSqlNewConn(s, false);
			return b;
			
		}
		
		
		//** Generate Index Queries
		public int cnt_PK110(string ArchiveID)
		{
			
			int B = 0;
			string TBL = "ArchiveHist";
			string WC = "Where ArchiveID = \'" + ArchiveID + "\'";
			
			B = DB.iGetRowCount(TBL, WC);
			
			return B;
		} //** cnt_PK110
		
		//** Generate Index ROW Queries
		public SqlDataReader getRow_PK110(string ArchiveID)
		{
			
			SqlDataReader rsData = null;
			string TBL = "ArchiveHist";
			string WC = "Where ArchiveID = \'" + ArchiveID + "\'";
			
			rsData = DB.GetRowByKey(TBL, WC);
			
			if (rsData.HasRows)
			{
				return rsData;
			}
			else
			{
				return null;
			}
		} //** getRow_PK110
		
		/// Build Index Where Caluses
		///
		public string wc_PK110(string ArchiveID)
		{
			
			string WC = "Where ArchiveID = \'" + ArchiveID + "\'";
			
			return WC;
		} //** wc_PK110
		
		//** Generate the SET methods
		
	}
	
}

File: SearchSVC/clsARCHIVEHIST.vb
Imports Microsoft.VisualBasic
Imports System.Configuration
Imports System
Imports System.Data.SqlClient
Imports System.Collections
Imports System.IO
Imports System.Data.Sql
 
Public Class clsARCHIVEHIST

    '** DIM the selected table columns 
    Dim DB As New clsDatabaseSVR
    Dim DMA As New clsDmaSVR
    Dim LOG As New clsLogging
    Dim UTIL As New clsUtilitySVR

    Dim ArchiveID As String = ""
    Dim ArchiveDate As String = ""
    Dim NbrFilesArchived As String = ""
    Dim UserGuid As String = ""

    Dim SecureID As Integer = -1

    Sub New(ByVal iSecureID As Integer)
        SecureID = iSecureID
    End Sub

    '** Generate the SET methods 
    Public Sub setArchiveid(ByRef val as string)
        If val$.Length = 0 Then
            val = Guid.NewGuid.ToString
        End If
        If Len(val) = 0 Then
            MsgBox("SET: Field 'Archiveid' cannot be NULL.: 2300")
            Return
        End If
        val = UTIL.RemoveSingleQuotes(val)
        ArchiveID = val
    End Sub

    Public Sub setArchivedate(ByRef val as string)
        val = UTIL.RemoveSingleQuotes(val)
        ArchiveDate = val
    End Sub

    Public Sub setNbrfilesarchived(ByRef val as string)
        If Len(val) = 0 Then
            val = "null"
        End If
        val = UTIL.RemoveSingleQuotes(val)
        NbrFilesArchived = val
    End Sub

    Public Sub setUserguid(ByRef val as string)
        val = UTIL.RemoveSingleQuotes(val)
        UserGuid = val
    End Sub



    '** Generate the GET methods 
    Public Function getArchiveid() As String
        If Len(ArchiveID) = 0 Then
            MsgBox("GET: Field 'Archiveid' cannot be NULL: 2400.")
            Return ""
        End If
        Return UTIL.RemoveSingleQuotes(ArchiveID)
    End Function

    Public Function getArchivedate() As String
        Return UTIL.RemoveSingleQuotes(ArchiveDate)
    End Function

    Public Function getNbrfilesarchived() As String
        If Len(NbrFilesArchived) = 0 Then
            NbrFilesArchived = "null"
        End If
        Return NbrFilesArchived
    End Function

    Public Function getUserguid() As String
        Return UTIL.RemoveSingleQuotes(UserGuid)
    End Function



    '** Generate the Required Fields Validation AS string 
    Public Function ValidateReqData() As Boolean
        If ArchiveID.Length = 0 Then Return false
        Return True
    End Function


    '** Generate the Validation AS string 
    Public Function ValidateData() As Boolean
        If ArchiveID.Length = 0 Then Return false
        Return True
    End Function


    '** Generate the INSERT AS string 
    Public Function Insert() As Boolean
        Dim b As Boolean = false
        Dim s As String = ""
        s = s + " INSERT INTO ArchiveHist("
        s = s + "ArchiveID,"
        s = s + "ArchiveDate,"
        s = s + "NbrFilesArchived,"
        s = s + "UserGuid) values ("
        s = s + "'" + ArchiveID + "'" + ","
        s = s + "'" + ArchiveDate + "'" + ","
        s = s + NbrFilesArchived + ","
        s = s + "'" + UserGuid + "'" + ")"
        Return DB.DBExecuteSql(SecureID, s, False)

    End Function


    '** Generate the UPDATE AS string 
    Public Function Update(ByVal WhereClause as string) As Boolean
        Dim b As Boolean = False
        Dim s As String = ""

        If Len(WhereClause) = 0 Then Return False

        s = s + " update ArchiveHist set "
        s = s + "ArchiveID = '" + getArchiveid() + "'" + ", "
        s = s + "ArchiveDate = '" + getArchivedate() + "'" + ", "
        s = s + "NbrFilesArchived = " + getNbrfilesarchived() + ", "
        s = s + "UserGuid = '" + getUserguid() + "'"
        WhereClause = " " + WhereClause
        s = s + WhereClause
        Return DB.DBExecuteSql(SecureID, s, False)
    End Function


    '** Generate the SELECT AS string 
    Public Function SelectRecs(ByVal SecureID As Integer) As SqlDataReader
        Dim b As Boolean = False
        Dim s As String = ""
        Dim rsData As SqlDataReader
        s = s + " SELECT "
        s = s + "ArchiveID,"
        s = s + "ArchiveDate,"
        s = s + "NbrFilesArchived,"
        s = s + "UserGuid "
        s = s + " FROM ArchiveHist"

        Dim CS$ = DBgetConnStr() : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader()
        Return rsData
    End Function


    '** Generate the Select One Row AS string 
    Public Function SelectOne(ByVal SecureID As Integer, ByVal WhereClause as string) As SqlDataReader
        Dim b As Boolean = False
        Dim s As String = ""
        Dim rsData As SqlDataReader
        s = s + " SELECT "
        s = s + "ArchiveID,"
        s = s + "ArchiveDate,"
        s = s + "NbrFilesArchived,"
        s = s + "UserGuid "
        s = s + " FROM ArchiveHist"
        s = s + WhereClause

        Dim CS$ = DBgetConnStr() : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader()
        Return rsData
    End Function


    '** Generate the DELETE AS string 
    Public Function Delete(ByVal WhereClause as string) As Boolean
        Dim b As Boolean = False
        Dim s As String = ""

        If Len(WhereClause) = 0 Then Return False

        WhereClause = " " + WhereClause

        s = " Delete from ArchiveHist"
        s = s + WhereClause

        b = DB.DBExecuteSql(SecureID, s, False)
        Return b

    End Function


    '** Generate the Zeroize Table AS string 
    Public Function Zeroize() As Boolean
        Dim b As Boolean = False
        Dim s As String = ""

        s = s + " Delete from ArchiveHist"

        b = DB.DBExecuteSql(SecureID, s, False)
        Return b

    End Function


    '** Generate Index Queries 
    Public Function cnt_PK110(ByVal ArchiveID As String) As Integer

        Dim B As Integer = 0
        Dim TBL$ = "ArchiveHist"
        Dim WC$ = "Where ArchiveID = '" + ArchiveID + "'"

        B = DB.iGetRowCount(TBL$, WC)

        Return B
    End Function     '** cnt_PK110

    '** Generate Index ROW Queries 
    Public Function getRow_PK110(ByVal SecureID As Integer, ByVal ArchiveID As String) As SqlDataReader

        Dim rsData As SqlDataReader = Nothing
        Dim TBL$ = "ArchiveHist"
        Dim WC$ = "Where ArchiveID = '" + ArchiveID + "'"

        rsData = DB.GetRowByKey(SecureID, TBL$, WC)

        If rsData.HasRows Then
            Return rsData
        Else
            Return Nothing
        End If
    End Function     '** getRow_PK110

    ''' Build Index Where Caluses 
    '''
    Public Function wc_PK110(ByVal ArchiveID As String) As String

        Dim WC$ = "Where ArchiveID = '" + ArchiveID + "'"

        Return WC
    End Function     '** wc_PK110

    '** Generate the SET methods 

End Class
